core/src/test/java/org/apache/accumulo/core/spi/compaction/DefaultCompactionPlannerTest.java

```diff
@@ -239,6 +243,38 @@ public void testMaxSize() {
     assertEquals(CompactionExecutorIdImpl.externalId("large"), job.getExecutor());
   }
 
+  /**
+   * Tests max.open can be overridden with maxOpen.
+   */
+  @Test
+  @SuppressWarnings("removal")
+  public void testOverrideMaxOpen() {
+    DefaultCompactionPlanner planner = new DefaultCompactionPlanner();
+    ConfigurationCopy aconf = new ConfigurationCopy(DefaultConfiguration.getInstance());
+    aconf.set(Property.TSERV_MAJC_THREAD_MAXOPEN.getKey(), "5");
+
+    ConfigurationImpl config = new ConfigurationImpl(aconf);
+
+    ServiceEnvironment senv = EasyMock.createMock(ServiceEnvironment.class);
+    EasyMock.expect(senv.getConfiguration()).andReturn(config).anyTimes();
+    EasyMock.replay(senv);
+
+    String executors = getExecutors("'type': 'internal','maxSize':'32M','numThreads':1",
+        "'type': 'internal','maxSize':'128M','numThreads':2", "'type': 'external','queue':'q1'");
+
+    // Test with max.open property
+    planner.init(getInitParams(senv, executors, "15"));
+    assertEquals(5, planner.getMaxFilesToCompact());
```

Comment:
If this works it would test a bit more and would not need to add the getMaxFilesToCompact() method

```suggestion
    var all = createCFs("F1", "1M", "F2", "1M", "F3", "1M", "F4", "1M", "F5", "11M","F6","1M","F7","1M");
    var params = createPlanningParams(all, all, Set.of(), 2, CompactionKind.SYSTEM);
    var plan = planner.makePlan(params);
    // TODO verify job has 5 files
```

Reply:
I ended up refactoring the compaction tests in DefaultCompactionPlannerTest to use a default conf and then set the property prior to compacting for tests that checked the plan files.

This cuts down on the amount of custom test code we were using while also getting these tests closer to mimicking a real-world situation.

Reply:
> while also getting these tests closer to mimicking a real-world situation.

That is nice.